### DISCUSSION TITLE
SQLiteGameDb plugin and UE5.2

### DISCUSSION THREAD

**Discussion by Ysgrathe** [Ysgrathe/SQLiteGameDb#4] [Announcements]
UPDATE:
I can report that the plugin compiles and works fine on 5.2, using the latest code.
It was probably ok before, but I have been doing some updates to it anyway, adding a couple of new features, and trying to namespace the sqlitecore from Epic so that it doesnt cause collisions with other plugins that require sqlitecore (it should now be using its own updated version 'SqliteCoreX'

Anyway, everything compiles fine on 5.2. I built and tested a SHIPPING build of the tutorial project and all is working as expected. 

NOTE: Running 'Standalone Game' from the editor does NOT work.
I'm not really sure why, and right now I don't really care to investigate, because running PIE works fine, and packaging for shipping also works fine, so it is more than likely a problem caused by it not being able to access the database assets (.db files) in the weird in-between land of standalone game mode.

**Comment by AbderrahimJami** [comment]
@Ysgrathe Hello, just wanted to thank you for making this plugin and explaining the process of using it in your YouTube series, I've been following along and it looks like a really good robust SQLite integration plugin. 

I've been trying to get it to work with my Unreal Engine version (5.3) unfortunately as soon as I try opening the .uproject file a window pops up asking me to rebuilt the SqliteGameDB and the SqliteCoreX modules using my IDE. I'm unable to downgrade to UE5.1/2 and I've tried rebuilding the plugin using the RunUAT.bat and passing the .uplugin to repackage it but it fails with loads of errors. 

If I just open the .sln on Rider it just complains about my MSBuildTool and says it's unable to load both the UE5 and the SqliteGameDB projects (I'm using the [recommended MSBuildTools for my UE5.3 version using VS22](https://dev.epicgames.com/documentation/en-us/unreal-engine/setting-up-visual-studio-development-environment-for-cplusplus-projects-in-unreal-engine#:~:text=14.38.33130-,14.38.33130,-Windows%20SDK)). 

I'm not entirely sure what's causing the build process to fail to be honest, and I don't know it's caused by the SQLite Third Library Include or something like that. I'm not very knowledgeable around using Third Party libraries in Unreal Engine but I was wondering if there's any plans for trying to get this plugin support added for newer versions of UE5. I'd be grealy appreciative if you could help in any way as I'd like to use your tool in my UE5 project and currently I can't seem to find anything better.

Cheers.

**Reply by Ysgrathe** [reply]
Hi there, sorry for the late reply didnt realise I had a message!
At this point, im finding it very difficult to recommend doing the SQLite thing in exaclty the same way I did.
The problem is this: EPIC have a SQLite plugin, and they dont update it to the latest versions of SQLite, so it lacks much of the very useful functionality available now.
I apptempted to get around this by making SQLiteCoreX which is nothing more than a clone of the Epic code, but with the newer SQLite included.
However, this makes issues, because even if I do that, it is possible for other modules and plugins to attempt to also include epics SQliteCore and then things get messy (you can either have my version or theirs, not both)
This is made worse by the fact that a few plugins by epic also make use of THEIR version, further stopping using my one.

TLDR: I no longer have any problems with this, since I started building unreal engine from source.
It enables you to insert the newer SQLite code, and have that built directly into the epic SQLiteCore - and MY plugin just uses that
If you are in any way familiar/confident with unreal and C++ I highly recomment building from source, it opens up a lot of useful possibilities and makes several tasks simpler/easier

**Reply by AbderrahimJami** [reply]
Hey, thanks for getting back to me! No worries about the late reply.

I actually managed to get your plugin working — I was trying to open your UE5.2 project, but it kept throwing build errors with the .sln. Ended up just rebuilding the plugin on its own and then exported that into my 5.3 project — and so far, it seems to be working fine! Fingers crossed I don’t hit any nasty errors down the line, but so far, so good.

Totally get where you’re coming from with Epic’s handling of the SQLite plugin — it does sound like a bit of a mess when different versions start clashing. Building the engine from source definitely seems like the way to go for stuff like this…

Anyway, thanks again for the insight — really appreciate it!
